docs/concepts/deployments.md

```diff
@@ -19,49 +18,78 @@ tags:
 
+The following diagram provides a high-level overview of the conceptual elements involved in defining a deployment and executing a flow run based on that deployment.
+
+```mermaid
+graph LR
+    F(Flow Code):::yellow -.-> A(Deployment Definition):::gold
+    subgraph Server [Prefect API]
+    D(Deployment):::green
+    end
+    subgraph Storage
+    B(Flow):::yellow
+    end
+    subgraph Infrastructure
+    G(Flow Run):::blue
+    end
+    A --> D
+    D --> E(Agent):::red
+    B -.-> E
+    A -.-> B
+    E --> G
+
+    classDef gold fill:goldenrod,stroke:goldenrod,stroke-width:4px
+    classDef yellow fill:gold,stroke:gold,stroke-width:4px
+    classDef gray fill:lightgray,stroke:lightgray,stroke-width:4px
+    classDef blue fill:blue,stroke:blue,stroke-width:4px,color:white
+    classDef green fill:green,stroke:green,stroke-width:4px,color:white
+    classDef red fill:red,stroke:red,stroke-width:4px,color:white
+    classDef dkgray fill:darkgray,stroke:darkgray,stroke-width:4px,color:white
+```
+
 When creating a deployment, a user must answer *two* basic questions:
 
-- What instructions does the agent need to set up an execution environment for my workflow? For example, a workflow may have Python requirements, unique Kubernetes settings, or Docker networking configuration.
+- What instructions does an [agent](/concepts/work-queues/) need to set up an execution environment for my workflow? For example, a workflow may have Python requirements, unique Kubernetes settings, or Docker networking configuration.
 - Where and how can the agent access the flow code?
 
 A deployment additionally enables you to:
 
-- Schedule flow runs
-- Assign tags for filtering flow runs on work queues and in the Prefect UI
-- Assign custom parameter values for flow runs based on the deployment
-- Create ad-hoc flow runs from the API or Prefect UI
-- Upload flow files to a defined storage location for retrieval at run time
+- Schedule flow runs.
+- Assign a work queue name to delegate deployment flow runs to work queues.
+- Assign one or more tags to organize your deployments and flow runs. You can use those tags as filters in the [Prefect UI](/ui/flow-runs/).
+- Assign custom parameter values for flow runs based on the deployment.
+- Create ad-hoc flow runs from the API or Prefect UI.
+- Upload flow files to a defined storage location for retrieval at run time.
+- Specify run time infrastructure for flow runs, such as Docker or Kubernetes configuration.
 
-Deployments can package your flow code; either Prefect Cloud or a local Prefect Orion server run with `prefect orion start`. 
+With storage blocks, you can package not only your flow code script but also any supporting files, including your custom modules, SQL scripts and any configuration files needed in your project.
```

Comment:
```suggestion
With remote storage blocks, you can package not only your flow code script but also any supporting files, including your custom modules, SQL scripts and any configuration files needed in your project.
```

In an effort to send the right signal with our names, I've been trying to replace conversational references to "storage" with "remote storage" to emphasize that they are only useful in some sort of distributed / remote context.